installers/msi-language/ActiveState/Network.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using Microsoft.Win32;
+using Microsoft.Deployment.WindowsInstaller;
+
+
+namespace ActiveState
+{
+    public class Network
+    {
+        private const string networkErrorKey = "NetworkError";
+        private const string networkErrorMessageKey = "NetworkErrorMessage";
+
+        /// <summary>
+        /// ResetErrorDetails clears the registry entries for network errors.
+        /// This function must be run from a deferred custom action
+        /// </summary>
+        public static void ResetErrorDetails(Session session)
+        {
+            // Deferred custom actions run as administrator so we have to specifically set
+            // the registry key for the user using their SID in order for the value to
+            // be available in later immediate custom actions
+            string registryKey = string.Format("HKEY_USERS\\{0}\\SOFTWARE\\ActiveState\\{1}", session.CustomActionData["USERSID"], session.CustomActionData["PRODUCT_NAME"]);
```

Comment:
Is `USERSID` still valid given @mdrohmann's latest changes?

Reply:
I believe that `USERSID` would still be the right choice here.

Reply:
Yes, the `USERSID` stuff is still required here so we can pass the network error details from the deferred custom action to an immediate one